subsys/settings/src/settings_nvs.c

```diff
@@ -183,6 +164,10 @@ static int settings_nvs_load(struct settings_store *cs,
 			break;
 		}
 	}
+
+#if CONFIG_SETTINGS_NVS_NAME_CACHE
+	cf->loaded = true;
```

Comment:
Note that `settings_call_set_handler` may return 1 (for instance, the user callback may decide not to continue loading more keys), in which case the cache may still not be fully populated. Why not to populate the cache at the initialization to be sure that all keys are iterated over?